glamod_marine_processing/qc_suite/scripts/next_level_marine_qc.py

```diff
@@ -0,0 +1,418 @@
+
+# read in SST stdev climatology
+at_stdev_clim_file = (
+    Path(data_dir) / "QCClimatologies" / "HadSST2_pentad_stdev_climatology.nc"
+)
+at_stdev_clim = Climatology.open_netcdf_file(
+    at_stdev_clim_file, "sst", time_axis="time"
+)
+
+print("Read SST temperature stdev clim")
+print(datetime.datetime.now())
+
+sst_anomalies = obs_sst.apply(
+    lambda row: calculate_anomalies(
+        value=row["observation_value"],
+        climatology=sst_clim,
+        lat=row["latitude"],
+        lon=row["longitude"],
+        date=row["date_time"],
+    ),
+    axis=1,
+)
+
+print("Calculated anomalies")
+print(datetime.datetime.now())
+
+groups = header.groupby("primary_station_id", group_keys=False)
+results = groups.apply(
+    lambda track: pd.Series(
+        do_track_check(
+            vsi=track["station_speed"],
+            dsi=track["station_course"],
+            lat=track["latitude"],
+            lon=track["longitude"],
+            date=track["report_timestamp"],
+            ids=track.name,
+            max_direction_change=60.0,
+            max_speed_change=10.0,
+            max_absolute_speed=40.0,
+            max_midpoint_discrepancy=150.0,
+        )
+    ),
+    include_groups=False,
+)
+results = results.explode()
+results.index = header.index
+track_check_qc = results.values.astype(int)
+
+print("Run track checks")
+print(datetime.datetime.now())
+
+# Buddy check
+limits = [[1, 1, 2], [2, 2, 2], [1, 1, 4], [2, 2, 4]]
+number_of_obs_thresholds = [[0, 5, 15, 100], [0], [0, 5, 15, 100], [0]]
+multipliers = [[4.0, 3.5, 3.0, 2.5], [4.0], [4.0, 3.5, 3.0, 2.5], [4.0]]
+
+selection = ~np.isnan(sst_anomalies)
+lats = obs_sst["latitude"].values[selection]
+lons = obs_sst["longitude"].values[selection]
+dates = obs_sst["date_time"][selection].reset_index().date_time
+
+buddy_check = mds_buddy_check(
+    lats,
+    lons,
+    dates,
+    sst_anomalies[selection],
+    at_stdev_clim,
+    limits,
+    number_of_obs_thresholds,
+    multipliers,
+)
+
+buddy_check_filled = np.zeros(len(sst_anomalies)) + 2.0
+buddy_check_filled[selection] = buddy_check[:]
+buddy_check = buddy_check_filled
+
+print("Ran buddy checks")
+print(datetime.datetime.now())
+
+
+position_check_qc = do_position_check(lat=header["latitude"], lon=header["longitude"])
+
+day_check = do_day_check(
+    date=header["report_timestamp"],
+    lat=header["latitude"],
+    lon=header["longitude"],
+)
+
+sst_climatology_check = do_climatology_check(
+    value=obs_sst["observation_value"],
+    climatology=sst_clim,
+    maximum_anomaly=8.0,  # K
+    lat=obs_sst["latitude"],
+    lon=obs_sst["longitude"],
+    date=obs_sst["date_time"],
+)
+
+sst_missing_check = do_missing_value_check(value=obs_sst["observation_value"])
+
+sst_freeze_check = do_sst_freeze_check(
+    sst=obs_sst["observation_value"],
+    freezing_point=271.35,
+    freeze_check_n_sigma=2.0
+)
+
+sst_hard_limit = do_hard_limit_check(
+    value=obs_sst["observation_value"],
+    limits=[268.15, 318.15],
+)
```

Comment:
See ['test_multiple_row_check'](https://github.com/ludwiglierhammer/glamod-marine-processing/blob/next_level_qc/tests/test_qc_on_db.py#L1713-L1805) using `apply_func=False`.
This gathers all required QC functions and run them at ones.